### PR TITLE
{bio}[foss/2019b] p4-phylogenetics v1.4-20210322 w/ Python 3.7.4

### PR DESCRIPTION
**Pull request by stevecdavis**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by migueldiascosta** [comment]
@boegelbot please test @ generoso

**Comment by stevecdavis** [comment]
> Test report by @boegelbot
> **FAILED**
> Build succeeded for 1 out of 2 (1 easyconfigs in total)
> generoso-c1-s-2 - Linux centos linux 8.2.2004, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz (haswell), Python 3.6.8
> See https://gist.github.com/96ac809200372289e04d01d21141ccc3 for a full test report.

I tried running --inject-checksums --force again and it came back with the same checksum that failed.

**Review by lexming — CHANGES_REQUESTED** [review]
I reviewed your PR and got P4 to build, but it was not trivial. Please check my PR https://github.com/stevecdavis/easybuild-easyconfigs/pull/2. It contains the following changes to this PR, which are needed to make it work:

* name changed to *p4-phylogenetics* - it's better to use the name of the github repo, because there is another p4 package in PYPI, which can be confusing
* sources changed to commit https://github.com/pgfoster/p4-phylogenetics/commit/d9c8d1b1036f4b52cd8c44d15855228c2df77da2 - Since there are no releases we have to pick a fixed commit to have reproducible code. I replaced your checkout of the master branch with the latest available commit. This is noted in the source code as version 1.4, so I set the easyconfig version to 1.4 plus the date of the commit. This is common practice for software without releases.
* added GSL and NLopt as dependencies - these are documented dependencies to build p4-phylogenetics
* added `prebuildopts` - the `sed` commands in `prebuildopts` redefine a couple of variables in `setup.py` with GSL paths to the libraries of EB
* disable `pip` - p4-phylogenetics is not compatible with it
* added `bin/p4` to the sanity checks

(note: merging https://github.com/stevecdavis/easybuild-easyconfigs/pull/2 will automatically update this PR)

File: easybuild/easyconfigs/p/p4-phylogenetics/p4-phylogenetics-1.4-20210322-foss-2019b-Python-3.7.4.eb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    ('NLopt', '2.6.1'),
+]
+
+use_pip = False
+download_dep_fail = True
+
+# Set paths to GSL from EasyBuild
+prebuildopts = 'sed -i "48d;s|^#my_lib_dirs.*|my_lib_dirs = [\'$EBROOTGSL/lib\']|" setup.py && '
+prebuildopts += 'sed -i "47d;s|^#my_include_dirs.*|my_include_dirs = [\'$EBROOTGSL/include\']|" setup.py && '
+
```
Comment (lexming) [inline easybuild/easyconfigs/p/p4-phylogenetics/p4-phylogenetics-1.4-20210322-foss-2019b-Python-3.7.4.eb:32]: Thanks for updating the PR. We just need this minor change to make the tests happy and I'll be able to submit a test report.
```suggestion

sanity_pip_check = True

```

**Review by lexming — CHANGES_REQUESTED** [review]

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node381.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/db1e6aecf1341f6cfcc68d18b8669dc9 for a full test report.

**Review by lexming — APPROVED** [review]
LGTM

**Comment by lexming** [comment]
Going in, thanks @stevecdavis !

**Pull request merged**